compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -1039,7 +1062,15 @@ impl<'tcx> SplitWildcard<'tcx> {
             // sometimes prefer reporting the list of constructors instead of just `_`.
             let report_when_all_missing = pcx.is_top_level && !IntRange::is_integral(pcx.ty);
             let ctor = if !self.matrix_ctors.is_empty() || report_when_all_missing {
-                Missing
+                let matrix_ctors = &self.matrix_ctors;
+                Missing(
+                    // Instead of calling `iter_missing` call directly and avoid
+                    // a `.cloned()`
```

Comment:
Is the `cloned` significant? `Constructor` should be quite cheap to clone.
But more importantly this is definitely where the perf impact is; we should avoid collecting constructors unless we know we'll use them. It's fine to have to rebuild a SplitWildcard later for diagnostics because most of time we don't emit diagnostics